include/gnc_containers.hpp

```diff
@@ -0,0 +1,60 @@
+//
```

Comment:
I already had this on the CommonSoftware side. The testing is a little lacking on my version though

Reply:
I'm working on porting my version to work with dynamically allocated memory in a stack structure. We need this for storing downlink packets.

Reply:
Let's talk integration at some point? It's unfortunate that we separately developed this.

Reply:
Upon further discussion with FSW we actually don't need the circular buffers for telemetry, so I'm getting rid of them. Yours is now the canonical implementation (which is good, since it actually works! :P)

Reply:
What do you mean to work with dynamically allocated memory in a stack structure?

Reply:
I mean that the size of the buffer is computed at runtime, but once it's computed the buffer size doesn't change.

For downlinking we originally wanted fixed-size FIFO buffers, which is what I mean by stack structure.